processing/src/main/java/org/apache/druid/query/JoinDataSource.java

```diff
@@ -108,10 +122,26 @@ public static JoinDataSource create(
       final DataSource right,
       final String rightPrefix,
       final JoinConditionAnalysis conditionAnalysis,
-      final JoinType joinType
+      final JoinType joinType,
+      final DimFilter leftFilter
+  )
+  {
+    return new JoinDataSource(left, right, rightPrefix, conditionAnalysis, joinType, leftFilter);
+  }
+
+  /**
+   * Create a join dataSource from an existing {@link JoinConditionAnalysis}.
```

Comment:
The javadoc doesn't seem right, it doesn't accept a JoinConditionAnalysis.

Fwiw, also, I don't really like adding "everything minus one" constructors when a new parameter is added. It minimizes the amount of code to update but it leaves things messier, and it makes it more likely that someone will forget to use the new parameter when it is actually important to use. IMO it is better to remove this constructor and update all the call sites to use the new one.